vm-presets.md

```diff
@@ -0,0 +1,293 @@
+applied.
+
+
+Conflicts
+------------------------
+
+`VirtualMachinePresets` use a similar conflict resolution strategy to
+Kubernetes `PodPresets`. If a portion of the domain spec is present in both a
+`VirtualMachine` and a `VirtualMachinePreset` and both resources have the
+identical information, then no error will occur and `VirtualMachine` creation
+will continue normally. If however there is a conflict between the resources,
+an error will occur and the `VirtualMachine` will not be created. For example:
+If both the `VirtualMachine` and `VirtualMachinePreset` define a `Volume`, but
+use different paths for the same name, KubeVirt will note the conflict.
+
+In the event that a conflict occurs, KubeVirt will create an `Event`.
+`kubectl describe` can be used to show all `Events`. For example:
+
+```
+$ kubectl describe ...
+....
+Events:
+  FirstSeen                         LastSeen                        Count From                              SubobjectPath                Reason    Message
+  Thu, 11 Jan 2018 08:34:14 -0500   Thu, 11 Jan 2018 08:34:14 -0500 1     {vmpreset.admission.kubernetes.io/vmpreset-allow-database }    conflict  Conflict on virtual machine preset. Duplicate mountPath /cache.
+
+```
+
+Because `VirtualMachinePresets` are implemented as an Initializer within the
+`virt-controller` pod, log messages associated with resource conflicts will
+also be reflected there.
+
+
+Creation and Usage
+------------------------
+
+`VirtualMachinePresets` are namespaced resources, so should be created in the
+same namespace as the `VirtualMachines` that will use them:
+
+`kubectl create -f <preset>.yaml [--namespace <namespace>]`
+
+KubeVirt will determine which `VirtualMachinePresets` apply to a Particular
+`VirtualMachine` by matching `Labels`. For example:
+
+```yaml
+kind: VirtualMachinePreset
+metadata:
+  name: example-preset
+  selector:
+    matchLabels:
+      flavor: foo
+  ...
+```
+
+would match any `VirtualMachine` in the same namespace with a `Label` of
+`flavor: foo`. For example:
+
+```yaml
+kind: VirtualMachine
+version: v1
+metadata:
+  name: myvm
+  labels:
+    flavor: foo
+  ...
+```
+
+
+Examples
+=============================
+
+Simple `VirtualMachinePreset` Example
+------------------------
+
+```yaml
+kind: VirtualMachinePreset
+version: v1alpha1
+metadata:
+  name: example-preset
+  selector:
+    matchLabels:
+      flavor: default-features
+spec:
+  domain:
+    features:
+      acpi: {}
+      apic: {}
+      hyperv:
+        relaxed: {}
+        vapic: {}
+        spinlocks:
+          spinlocks: 8191
+---
+kind: VirtualMachine
+version: v1
+metadata:
+  name: myvm
+  labels:
+    flavor: default-features
+spec:
+  firmware:
+    UUID: c8f99fc8-20f5-46c4-85e5-2b841c547cef
+```
+
+Once the `VirtualMachinePreset` is applied to the `VirtualMachine`, the
+resulting resource would look like this:
+
+
+```yaml
+kind: VirtualMachine
+version: v1
+metadata:
+  name: myvm
+  labels:
+    flavor: windows-10
```

Comment:
I'm  a bit confused,   where did the "windows-10" flavor came from ? 
above i can see     flavor: default-features  both in the preset and the VM.

Reply:
You are correct. They should match. Alas, the dangers of creating examples by hand.